### PR TITLE
RELATED: RAIL-3849 Resolve filtering by parents in child attribute filter

### PR DESCRIPTION
**Pull request by plavemar**
- handle element filtering after parent attribute filter changed
- add all possible errors to onError callback

JIRA: RAIL-3849

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx [hunk 1]
```diff
@@ -344,6 +345,17 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
 
     const resolvedParentFilters = useResolveValueWithPlaceholders(props.parentFilters);
 
+    useEffect(() => {
+        if (!isEqual(prevParentFilters, props.parentFilters)) {
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:349]: What if there are no parent filters?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5935/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7715/ : SUCCESS in 23m 07s

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx (hunk 1 above)
Comment (plavemar) [reply libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx]: Added parentFilter availability check

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5936/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7716/ : SUCCESS in 17m 24s

**Comment by plavemar** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dashboard.spec.ts**|9|-|16.4170s|
|✅|**drilling.spec.ts**|2|-|10.1040s|
|✅|**dateFilter.spec.ts**|7|-|14.2760s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/70/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5937/ : SUCCESS in 30s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/70/ : SUCCESS in 8m 15s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1967

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1710/ : SUCCESS in 1m 29s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1991/ : SUCCESS in 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/788/ : SUCCESS in 18m 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7717/ : SUCCESS in 18m 20s

**Pull request merged**